client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -254,38 +258,42 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder) {
 
     @Override
     public CompletableFuture<Boolean> scaleStream(final Stream stream, final List<Integer> sealedSegments,
+                                                  final Map<Double, Double> newKeyRanges, final long timeoutMillis,
```

Comment:
One suggestion is that we provide all three calls: `scaleStream`, `startScale`, `checkScale`.

`scaleStream` can be a long poll, whereas `startScale` and `checkScale` can be left as they are. For  `startScale` and `checkScale`, we tell the developer that the code needs to check periodically, but the logic that you are trying to introduce here is left to the application to implement.